lib/logger/lib/logger.ex

```diff
@@ -489,8 +489,13 @@ defmodule Logger do
           level: min_level, utc_log: utc_log?} = Logger.Config.__data__
 
         if compare_levels(level, min_level) != :lt do
-          truncated = truncate(chardata_or_fn, truncate)
-          metadata = [pid: self()] ++ Keyword.merge(pdict, metadata)
+          {message, fn_metdata} = normalize_message(chardata_or_fn)
```

Comment:
Can we invert the flow of control here? Instead of:

```elixir
{message, fn_metdata} = normalize_message(chardata_or_fn)
metadata = ...
```

We could do:

```elixir
metadata = [pid: self()] ++ Keyword.merge(pdict, metadata)
{message, metadata} = normalize_message(chardata_or_fn, metadata)
```

This way we will only merge the function metadata if there is function metadata. :)